core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -390,7 +390,7 @@ class ReplicaManager(val config: KafkaConfig,
               // epoch, a sentinel value (NoEpoch) is used and bypass the epoch validation.
               if (requestLeaderEpoch == LeaderAndIsr.EpochDuringDelete ||
                   requestLeaderEpoch == LeaderAndIsr.NoEpoch ||
-                  requestLeaderEpoch > currentLeaderEpoch) {
+                  requestLeaderEpoch >= currentLeaderEpoch) {
```

Comment:
Just curious -- was the epoch check put in place because we were concerned about stale stop replicas? Just trying to figure out why we need it and the implications for adding the current epoch.

Reply:
Looks like this is part of KIP-570. I will take a look.

Reply:
Seems like the concern there was reassignment. I think equal to the leader epoch is ok though because if we are a replica for the current leader epoch then we can't send a stop replica unless the reassignment was cancelled?

Reply:
I think the main thing is that a new reassignment will always have a leader epoch bump, so we are still ensured that the `StopReplica` cannot be reordered. It is still possible to have a stray replica though if the `StopReplica` from a cancellation is received before the initial `LeaderAndIsr` when the reassignment began. A better fix would be to bump the epoch after cancellation, but I didn't see a really simple way to do that.

Reply:
Got it. Thanks for clarifying.